website/docs/plugin/framework/deprecations.mdx

```diff
@@ -0,0 +1,1282 @@
+
+-> [Required attributes](/terraform/plugin/framework/handling-data/schemas#required) are also referred to as required "arguments" throughout the Terraform documentation.
+
+In general, the procedure here does two things:
+
+- Prevents the operator from needing to define two attributes with the same value.
+- Allows the operator to migrate the configuration to the new attribute at the same time requiring that any other references only work with the new attribute. This is to prevent a situation with Terraform showing a difference when the existing attribute is configured, but the new attribute is saved into the Terraform state. For example, in `terraform plan` output format:
+
+```
+existing_attribute: "" => "value"
+new_attribute:      "value" => ""
+```
+
+The recommended process is as follows:
+
+1. Replace `Required: true` with `Optional: true` in the existing attribute schema definition.
+1. Replace `Required` with `Optional` in the existing attribute documentation.
+1. Duplicate the schema definition of the existing attribute, renaming one of them with the new attribute name.
+1. Duplicate the documentation of the existing attribute, renaming one of them with the new attribute name.
+1. Add a [`DeprecationMessage`](/terraform/plugin/framework/handling-data/schemas#deprecationmessage-1) to the schema definition of the existing (now the "old") attribute, noting to use the new attribute in the message.
+1. Add `**Deprecated**` to the documentation of the existing (now the "old") attribute, noting to use the new attribute.
+1. Add a note to the documentation that either the existing (now the "old") attribute or new attribute must be configured.
+1. Add the type-specific [validator](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework-validators) `{type}validator.ExactlyOneOf` to the schema definition of the new attribute, with a path expression matching the old attribute. This will ensure at least one of the attributes is configured, but present an error to the operator if both are configured at the same time. For example, an attribute of type string would use the [`stringvalidator.ExactlyOneOf`](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator#ExactlyOneOf) validator.
+1. Add conditional logic in the `Create` and `Update` functions of the data source or resource to handle both attributes. Generally, this involves using [`{type}.IsNull()`](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/types/basetypes).
+1. Follow the rest of the procedures in the [Provider Attribute Removal section](#provider-attribute-removal). When the old attribute is removed, update the schema definition and documentation of the new attribute back to `Required`, and remove the `{type}validator.ExactlyOneOf` validator.
+
+#### Example Renaming of a Required Attribute
+
+Given this sample resource:
+
+```go
+package provider
```

Comment:
Not sure if I correctly transcribed the original SDK deprecation code examples over, this is a first attempt 😄